model/sensor.go

```diff
@@ -58,6 +58,33 @@ const (
 	keyCustom    = "custom"
 )
 
+// The keys for sensor formats.
+// See also SensorFormats below.
+const (
+	keyBool    string = "bool"
```

Comment:
Any reason why you're not having these typed ?

Reply:
This would require the sensor type itself to change, or mean that we would be calling ```string(keyBool)`` very often. I'm open to other suggestions if I have missed something

Reply:
I think we should do this and just use a shim between this and the device if we have to (or just string). Maybe in future we can change Device to use this type.

Reply:
I have taken a stab at implementing a shim.

Changing the actual fields of the sensorV2 struct is a very big change which certainly warrants a PR in itself (and lots of testing). I am not convinced that change should block this change.

Reply:
IMO, datastore entities typically use standard datastore types, which tend to be elemental GoLang types.

Reply:
Is that a technical limitation ? Or can it flatten things out to basic types e.g. would ``` type Input string``` just flatten out to string in the datastore ?